daisy_workflows/image_import/import_image.sh

```diff
@@ -48,22 +53,48 @@ function resizeDisk() {
   fi
 }
 
-# Mount GCS bucket containing the disk image.
-mkdir -p /gcs/${SOURCEBUCKET}
-gcsfuse --implicit-dirs ${SOURCEBUCKET} /gcs/${SOURCEBUCKET}
+function copyImageToScratchDisk() {
+  local sourceSizeGigabytes="$((1 + SOURCE_SIZE_BYTES / 1024 / 1024 / 1024))"
+  local scratchDiskSizeGigabytes=sourceSizeGigabytes
+  # We allocate double capacity for OVA, which would
+  # require making an additional copy of its enclosed VMDK.
+  if [[ "${SOURCE_FILE_TYPE}" == "ova" ]]; then
+    scratchDiskSizeGigabytes="$((sourceSizeGigabytes * 2))"
+  fi
+
+  # This disk is initially created with 10GB of space.
+  # Enlarge it if that's insufficient to hold the input image.
+  if [[ ${scratchDiskSizeGigabytes} -gt 10 ]]; then
+    resizeDisk "${SCRATCH_DISKNAME}" "${scratchDiskSizeGigabytes}" "${ZONE}"
+  fi
+
+  local scratchPath="/daisy-scratch/$(dirname ${SOURCEPATH})/"
+  mkdir -p "${scratchPath}"
+  # /dev/sdb is used since the scratch disk is the second
+  # disk that's attached in import_disk.wf.json.
+  mkfs.ext4 /dev/sdb
+  mount /dev/sdb "${scratchPath}"
+  if [[ $? -ne 0 ]]; then
+    echo "ImportFailed: Failed to prepare scratch disk."
+  fi
+  gsutil cp "${SOURCEURL}" "${scratchPath}"
+  echo "Import: Copied image from ${SOURCEURL} to ${scratchPath}"
+}
+
+copyImageToScratchDisk
 
-# Atrocious OVA hack.
-SOURCEFILE_TYPE="${$SOURCE_DISK_FILE##*.}"
-if [[ "${SOURCEFILE_TYPE}" == "ova" ]]; then
+# If the image is an OVA, then copy out its VMDK.
+if [[ "${SOURCE_FILE_TYPE}" == "ova" ]]; then
   echo "Import: Unpacking VMDK files from ova."
-  VMDK="$(tar --list -f /gcs/${SOURCEPATH} | grep -m1 vmdk)"
-  tar -C /gcs/${DAISY_SOURCE_URL#"gs://"} -xf /gcs/${SOURCEPATH} ${VMDK}
+  VMDK="$(tar --list -f /daisy-scratch/${SOURCEPATH} | grep -m1 vmdk)"
+  tar -C /daisy-scratch/${DAISY_SOURCE_URL#"gs://"} -xf /daisy-scratch/${SOURCEPATH} ${VMDK}
   SOURCEPATH="${DAISY_SOURCE_URL#"gs://"}/${VMDK}"
   echo "Import: New source file is ${VMDK}"
 fi
 
+# Ensure the output disk has sufficient space to accept the disk image.
 # Disk image size info.
-SIZE_BYTES=$(qemu-img info --output "json" /gcs/${SOURCEPATH} | grep -m1 "virtual-size" | grep -o '[0-9]\+')
+SIZE_BYTES=$(qemu-img info --output "json" /daisy-scratch/${SOURCEPATH} | grep -m1 "virtual-size" | grep -o '[0-9]\+')
  # Round up to the next GB.
 SIZE_GB=$(awk "BEGIN {print int((${SIZE_BYTES}/1000000000)+ 1)}")
```

Comment:
ref to export_disk_ext.sh about how to calculate size_gb
We may fix this by the way? Now it almost always grant extra spaces because 1GB=1073741824>1000000000

Reply:
done!